include/xpedite/util/AddressSpace.H

```diff
@@ -37,10 +37,10 @@ namespace xpedite { namespace util {
       Segment() = default;
 
       Segment(Pointer begin_, Pointer end_, bool canRead_, bool canWrite_, bool canExec_,
```

Comment:
All those boolean arguments to the constructor make it easy to invoke the constructor with wrong arguments (for example wrong order).
```c++
Segment s{start, end, true, false, true, false, true, file};
```
How do I know I've passed the right flags? Should we convert them to enums? This way it could be initialized like that:

```c++
Segment s{begin, end, ReadBit::Enabled, WriteBit::Enabled, ExecBit::Enabled, PositionIndependent::False, HugePage::True, path};
```
which I think makes it more readable and less error prone.

Reply:
Good point. Not sure about creating one type for each argument. These are attributes for the page backing the segment. may be we can define one enumeration with non-overlapping values and or them together in a bit set like below.

`
enum class SegmentAttributes {
  ReadEnabled = 1  << 0,
  WriteEnabled = 1 <<  1
  ...
};
`

We can then provide a `list` of those values as arguments.